### PR TITLE
German holidays/Correction for year_ranges with different regions

### PR DESCRIPTION
**Pull request by knut2**
See #174 for code changes.

Changes in Holidays:
- Add Ostersonntag, Pfingstsonntag, Heilig Abend and Silvester as informal holidays.
- Add Fronleichnam for de_sn_aux, de_th_aux
- Add Reformationstag as informal holiday for de_bw (it is school free)
- Add Reformationstag for 2017 (500 years)
- Correction in test for :de_by_aux

### CODE REVIEW

File: definitions/de.yaml [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # German holiday definitions for the Ruby Holiday gem.
 #
-# Updated: 2009-11-05.
+# Updated: 2016-03-16
+#
+# Changes 2016-03-16:
```
Comment (ptrimble) [inline definitions/de.yaml]: Excellent notes, thank you!

File: definitions/de.yaml [hunk 2]
```diff
@@ -11,18 +19,57 @@ months:
...
     regions: [de_bw, de_by, de_he, de_nw, de_rp, de_sl]
     function: easter(year)+60
+  - name: Fronleichnam
+    regions: [de_sn_aux, de_th_aux]
+    function: easter(year)+60
+  - name: Weiberfastnacht
+    regions: [de_bw, de_by, de_he, de_nw, de_rp, de_sl, de_sn ]
+    function: easter(year)-52
+    type: informal
+  #~ - name: Fastnachtfreitag
```
Comment (ptrimble) [inline definitions/de.yaml]: Are these not required anymore? Can we just remove the commented definitions?

**Comment by ptrimble** [comment]
@knut2 I would like those commented lines removed and the tests seem to be failing. I also see the coverage dropped but that could be because of the failed tests. Are you willing to make the updates or would you prefer that I finish up the final steps (with the commit attributed to you, of course). I offer because I would like to get through these PRs before merging my large refactor.

Let me know!

File: definitions/de.yaml (hunk 2 above)
Comment (knut2) [inline definitions/de.yaml]: Hello,
the commented definitions are "very informal" holidays, you can remove them,

**Comment by knut2** [comment]
Hello, 
> Are you willing to make the updates or would you prefer that I finish up the final steps
I would prefer if you can finish the steps.
I had problems to call the tests in my installation (the gemspec command `git ls-files` did not work and I needed workarounds)

Thanks for your work!

**Comment by ptrimble** [comment]
Not a problem! I'm happy to get us across the finish line. I'll do so this evening.

Thanks for the PR!

**Comment by ptrimble** [comment]
Closing this PR in favor of https://github.com/holidays/holidays/pull/183, which is this PR plus one additional commit to fix the failing build. I'll merge that one once the builds are green.

Thanks again!

**Pull request closed without merging**